Userland/Applications/Calculator/Calculator.cpp

```diff
@@ -20,10 +48,16 @@ Optional<Crypto::BigFraction> Calculator::operation_with_literal_argument(Operat
 
     // If a previous operation is still in progress, finish it
     // Makes hitting "1+2+3=" equivalent to hitting "1+2=+3="
-    if (m_binary_operation_in_progress != Operation::None) {
+    if (m_binary_operation_in_progress != Operation::None && !operation_is_unary(operation)) {
```

Comment:
I thought it was a bugfix but on second look it just creates more problems.

The issue is that if you're part way through making a binary expression, lets say you've entered "3*", hitting the sqrt operator would complete the current operation then apply sqrt to it, returning a 3 (since sqrt(3*3) is 3). This happens with any of the unary operators (Inverse, ToggleSign, Sqrt, Percent)

My change would just apply the sqrt to what is on the left side of the binary operation... and nothing else. 

I am going to assume the old behaviour is correct, and can fix it in a seperate PR 😀

Reply:
You can also do that, it's fine.
Welcome to the project btw!